crates/executor/src/executor.rs

```diff
@@ -371,46 +376,63 @@ where
+            // let (revm::ExecutionResult { exit_reason, gas_used, logs, .. }, state) = out;
 
             // Fatal internal error.
-            if exit_reason == revm::Return::FatalExternalError {
+            if let Err(revm::primitives::EVMError::Database(_)) = out {
                 return Err(Error::ExecutionFatalError)
             }
 
             // Success flag was added in `EIP-658: Embedding transaction status code in receipts`.
             // TODO for verification (exit_reason): some error should return EVM error as the block
             // with that transaction can have consensus error that would make block
             // invalid.
-            let is_success = match exit_reason {
-                revm::return_ok!() => true,
-                revm::return_revert!() => false,
-                _ => false,
-                // TODO: Handle after bumping to revm v3.0: https://github.com/paradigmxyz/reth/issues/463
-                // e => return Err(Error::EVMError { error_code: e as u32 }),
+
+            let is_success = if let Ok(ResultAndState { ref result, .. }) = out {
+                match &result {
+                    Success { .. } => true,
+                    _ => false,
+                }
+            } else {
+                false
             };
+            // let is_success = match exit_reason {
+            //     revm::return_ok!() => true,
+            //     revm::return_revert!() => false,
+            //     _ => false,
+            //     // TODO: Handle after bumping to revm v3.0: https://github.com/paradigmxyz/reth/issues/463
+            //     // e => return Err(Error::EVMError { error_code: e as u32 }),
+            // };
 
             // Add spent gas.
-            cumulative_gas_used += gas_used;
-
-            // commit state
-            let (changeset, new_bytecodes) = self.commit_changes(state);
-
-            // Transform logs to reth format.
-            let logs: Vec<Log> = logs.into_iter().map(into_reth_log).collect();
-
-            // Push transaction changeset and calculte header bloom filter for receipt.
-            changesets.push(TransactionChangeSet {
-                receipt: Receipt {
-                    tx_type: transaction.tx_type(),
-                    success: is_success,
-                    cumulative_gas_used,
-                    bloom: logs_bloom(logs.iter()),
-                    logs,
-                },
-                changeset,
-                new_bytecodes,
-            })
+            if let Ok(ResultAndState { ref result, .. }) = out {
+                cumulative_gas_used += match &result {
+                    Success { gas_used, .. } => gas_used,
+                    Revert { gas_used, .. } => gas_used,
+                    Halt { gas_used, .. } => gas_used,
+                }
+            };
+            // cumulative_gas_used += gas_used;
+
+            if let Ok(ResultAndState { result: Success { logs, .. }, state }) = out {
```

Comment:
Already made changes, just wanting to comment this to clear confusion, maybe it will be useful for future.
Initial discussion on revm returnables was here: https://github.com/bluealloy/revm/issues/309

tldr is: revm return type is specified as `Result<ResultAndState,EVMError>`.
Only if it is `EVMError`, transaction is invalid and whole block becomes invalid.

If it is Ok as in `ResultAndState`, transactions is valid and can be found inside block but `Receipt::is_success` needs to be set depending on `ExecutionResult`: If `Success` set true, if `Revert` or `Halt` return `false`